src/scripting/game_lua_kernel.cpp

```diff
@@ -1759,20 +1766,22 @@ int game_lua_kernel::intf_find_path(lua_State *L)
 		calc.reset(new lua_pathfind_cost_calculator(L, arg));
 	}
 
-	const team& viewing_team = viewing_side
-		? board().get_team(viewing_side)
-		: board().get_team(u->side());
-
 	pathfind::teleport_map teleport_locations;
 
 	if(!ignore_teleport) {
-		teleport_locations = pathfind::get_teleport_locations(*u, viewing_team, see_all, ignore_units);
+		if(viewing_side == 0) {
+			return luaL_error(L, "wesnoth.find_path: ignore_teleport=false requires a valid viewing_side");
+		} else {
+			teleport_locations = pathfind::get_teleport_locations(*u, board().get_team(viewing_side), see_all, ignore_units);
+		}
 	}
 
 	if (!calc) {
-		if (!u) return luaL_argerror(L, 1, "unit not found");
+		if(!u) {
+			return luaL_argerror(L, 1, "unit not found OR custom cost function not provided");
+		}
 
-		calc.reset(new pathfind::shortest_path_calculator(*u, viewing_team,
+		calc.reset(new pathfind::shortest_path_calculator(*u, board().get_team(viewing_side),
```

Comment:
This would throw if `viewing_side` were 0, wouldn't it?

Reply:
Yes, but it only reaches here if there's a valid unit, in which case `viewing_side` cannot be 0.

It's set in two places - if there's a unit, it's set to the side of that unit. Then if `viewing_side` is explicitly set in options, it will be used instead if it's valid.

Reply:
Alright, I see it now.  I was thinking that it could get here in the case of a provided location that doesn't have a unit on it, but in that case `!u` would be true and it would show the error instead.